### DISCUSSION TITLE
Please add CRT shaders

### DISCUSSION THREAD

**Discussion by harvald** [ihhub/fheroes2#8157] [Ideas]
Please consider adding CRT shaders like those in dosbox staging.
Example looks [https://krystof.io/dosbox-shaders-comparison-for-modern-dos-retro-gaming/](url)

**Comment by ihhub** [comment]
Hi @harvald , can you explain the reason why and for what it is needed? What other projects do might not be applicable to this one.

**Reply by harvald** [reply]
Those shaders generally improve looks of old, low resolution, pixel art games. Games graphics was developed on CRT monitors, and artists did, what looks best on slightly blurred screens. Especially today on super sharp and big monitors those pixels are somewhat too contrast. CRT shaders give specific effect which looks great in my opinion.

**Reply by ihhub** [reply]
This is a very subjective opinion as many players prefer sharp images rather than blurred ones. You can just set linear scaling and go to a full screen mode with resolution lower than your OS to see that the blurring is present. Especially, if you in-game resolution is not integer scaled to your OS resolution.

**Reply by harvald** [reply]
I used to prefer thinking this way myself. I didn't like scanlines, blurred scaling, etc, until I saw CRT shaders. I can't describe differences, but those are different.
If You decide it is not good for fheroes2, then OK. I love it anyway (Great work BTW ;) ). It's simply idea, not strong opinion.

**Comment by AlexSnowLeo** [comment]
> This is a very subjective opinion as many players prefer sharp images rather than blurred ones. You can just set linear scaling and go to a full screen mode with resolution lower than your OS to see that the blurring is present. Especially, if you in-game resolution is not integer scaled to your OS resolution.

In my opinion, it would be nice to see such an option in a project whose roots lead to the DOS version. After all, initially Heroes II was launched on CRT monitors under DOS. Nostalgia, so to speak.

Just for example:
https://www.youtube.com/watch?v=Q6_IfojvGEY

For example I recording gameplay of the original HMM II DOS game in DosBox Staging with default CRT shader. Please see in full screen:
https://youtu.be/LqQdmoHW5do

**Comment by minuspinus** [comment]
Very cool effect realized in the game: https://shotgun-king.com
![telegram-cloud-photo-size-2-5375488438570182424-y](https://github.com/user-attachments/assets/5df5ae42-d423-4bbb-b277-01710b0557c7)
![telegram-cloud-photo-size-2-5375488438570182422-y](https://github.com/user-attachments/assets/dfc498a9-f572-49d8-b6f5-3accf4148501)
![telegram-cloud-photo-size-2-5375488438570182421-y](https://github.com/user-attachments/assets/fd271fdd-4553-4542-a45e-aeae46e77de8)
